docs/spec/concepts.rst

```diff
@@ -3,6 +3,268 @@
+_some_ specific static type, but _which_ static type is not known. A static
+type checker should not emit static type errors on an expression or statement
+if :ref:`Any` might represent a static type which would avoid the error. (This
+intuition is made more precise below, in our definitions of materialization and
+assignability.)
+
+Similarly, a gradual type such as ``tuple[int, Any]`` (see :ref:`tuples`) or
+``int | Any`` (see :ref:`union-types`) does not represent a single set of
+Python objects; rather, it represents a (bounded) range of possible sets of
+values.
+
+In the same way that ``Any`` does not represent "the set of all Python objects"
+but rather "an unknown set of objects", ``tuple[int, Any]`` does not represent
+"the set of all length-two tuples whose first element is an integer." That is a
+fully static type, spelled ``tuple[int, object]``.  In contrast, ``tuple[int,
+Any]`` represents some unknown set of tuple values; it might be the set of all
+tuples of two integers, or the set of all tuples of an integer and a string, or
+some other set of tuple values.
+
+In practice, this difference is seen (for example) in the fact that we can
+assign an expression of type ``tuple[int, Any]`` to a target typed as
+``tuple[int, int]``, whereas assigning ``tuple[int, object]`` to ``tuple[int,
+int]`` is a static type error. (Again, we formalize this distinction in the
+below definitions of materialization and assignability.)
+
+In the same way that the fully static type ``object`` is the upper bound for
+the possible sets of values represented by ``Any``, the fully static type
+``tuple[int, object]`` is the upper bound for the possible sets of values
+represented by ``tuple[int, Any]``.
+
+The gradual guarantee
+~~~~~~~~~~~~~~~~~~~~~
+
+:ref:`Any` allows gradually adding static types to a dynamically-typed program.
+For a fully dynamically-typed program, a static checker has the type :ref:`Any`
+for all expressions, and should emit no errors. Adding static type annotations
+to the program (making the program more statically typed) may result in static
+type errors, if the program is not correct or if the static type annotations
+aren't able to fully represent the runtime types. Removing static type
+annotations (making the program more dynamic) should not result in additional
+static type errors. This is often referred to as the **gradual guarantee**.
+
+In Python's type system, we don't take the gradual guarantee as a strict
+requirement, but it's a useful guideline.
+
+Subtype, supertype, and type equivalence
+----------------------------------------
+
+A fully static type ``B`` is a **subtype** of another fully static type ``A``
+if and only if the set of values represented by ``B`` is a subset of the set of
+values represented by ``A``. Because the subset relation on sets is transitive
+and reflexive, the subtype relation is also transitive (if ``C`` is a subtype
```

Comment:
I feel like this edit implies that there is also a subtype relation on not-fully-static types, which is not true (and it's important that it's not true.)

Reply:
Right. This was another wordsmithing suggestion, for symmetry between the two halves of the sentence ("the subset relation on sets" becomes balanced by "the subtype relation on static types"). Maybe this?

```suggestion
and reflexive, the subtype relation on types is also transitive (if ``C`` is a subtype
```

Reply:
Ah, I see what you're going for. But I feel like "subtype relation on types" also risks implying that it is a relation on all types, including gradual ones.

I think this might be a case where the sentence symmetry is not as important as ensuring we don't imply anything inaccurate by adding unnecessary qualifiers.

Reply:
Makes sense.